src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovySandbox.java

```diff
@@ -121,7 +122,10 @@ public Scope enter() {
             if (ExtensionList.lookup(RootAction.class).get(ScriptApproval.class) == null) {
                 return; // running in unit test, ignore
             }
-            ScriptApproval.get().accessRejected(x, _context);
+            String signature = x.getSignature();
+            if (signature != null && !StaticWhitelist.isPermanentlyBlacklisted(signature)) {
```

Comment:
Uh, what? No. You need to call `accessRejected` here even when `signature == null` (that would be an “unclassified” error).